tests/test_agent_procedures.py

```
@@ -0,0 +1,138 @@
+def test_agents_md_top_pointer_present() -> None:
+    """AGENTS.md top must summarize the memory-first principle."""
+    text = (REPO_ROOT / "AGENTS.md").read_text(encoding="utf-8")
+    assert "Memory-first" in text, "AGENTS.md missing Memory-first top-level pointer"
+    assert "Tier-3 Substrate Query" in text or TIER3_SECTION_HEADING in text, (
+        "AGENTS.md memory-first section must reference the Tier-3 Substrate Query"
+    )
+    pointer_pos = text.find("Memory-first")
+    mandatory_pos = text.find("## Mandatory reading")
+    assert pointer_pos > 0 and mandatory_pos > pointer_pos, (
+        "AGENTS.md 'Memory-first' pointer must precede '## Mandatory reading'."
+    )
```

Ruff (0.15.12)

[warning] Assertion should be broken down into multiple parts

(PT018)